api/v1alpha1/overlaynetwork_types.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Important: Run "make" to regenerate code after modifying this file
+
+// NetworkAttachmentDefinitionRequestSpec defines the desired state of NetworkAttachmentDefinitionRequest
+type NetworkAttachmentDefinitionRequestSpec struct {
+	Name               string `json:"name"`
+	LogicalNetworkName string `json:"logicalNetworkName"`
+	Type               string `json:"type,omitempty"`
+}
+
+// OverlayNetworkStatus defines the observed state of OverlayNetwork
+type NetworkAttachmentDefinitionRequestStatus struct {
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// NetworkAttachmentDefinitionRequest is the Schema for the network attachment definition request API
+type NetworkAttachmentDefinitionRequest struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NetworkAttachmentDefinitionRequestSpec   `json:"spec,omitempty"`
+	Status NetworkAttachmentDefinitionRequestStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// NetworkAttachmentDefinitionRequestList contains a list of NetworkAttachmentDefinitionRequest
+type NetworkAttachmentDefinitionRequestList struct {
```

Comment:
Why do we need this list?

Reply:
I think it enables listing the CRD instances, for example `kubectl get <crd-name>`

Reply:
It was auto-generated before, I guess it serves listing the objects of this type?

Reply:
Yes, it was auto-generated. But maybe we don't need it. Would watching the CRD work without it?

Reply:
It is needed for listing all CR/s of a specific CRD, which sounds as a requirement for an informer.